core/src/Streamly/Internal/Data/Serialize/TH.hs

```diff
@@ -136,8 +144,185 @@ isUnitType _ = False
 -- Size
 --------------------------------------------------------------------------------
 
-mkSizeOfExpr :: TypeOfType -> Q Exp
-mkSizeOfExpr tyOfTy =
+{-
+int_w8 :: Int -> Word8
+int_w8 = fromIntegral
+-}
+
+w8_w16 :: Word8 -> Word16
+w8_w16 = fromIntegral
+
+w8_w32 :: Word8 -> Word32
+w8_w32 = fromIntegral
+
+w8_w64 :: Word8 -> Word64
+w8_w64 = fromIntegral
+
+{-
+w32_int :: Word32 -> Int
+w32_int = fromIntegral
+-}
+
+c2w :: Char -> Word8
+c2w = fromIntegral . ord
+
+shiftAdd :: Bits a => (b -> a) -> [b] -> a
+shiftAdd conv xs =
+    foldl' (.|.) zeroBits $
+    map (\(j, x) -> shiftL x (j * 8)) $ zip [0 ..] $ map conv xs
+
+-- Note: This only works in little endian machines
+xorCmp :: [Word8] -> Name -> Name -> Q Exp
+xorCmp tag off arr =
```

Comment:
Instead of generating this via TH can't we write it directly in Haskell and use that? Creating one comparison function for each deserialization may be too much code and may not be necessary.

Reply:
We can do that. I'm going to add this as a developer comment and open an issue for keeping track of these comments. These will be tested with a bigger project soon. We can fix these according to our experience with that.